devicetypes/smartthings/cooper-scene-controller.src/cooper-scene-controller.groovy

```diff
@@ -0,0 +1,268 @@
+	//cmd.value (0-31) is a binary representation of current switch state
+	//switch 1 - first bit
+	setSwitchState(1, (cmd.value & 1)? "on" : "off")
+	//switch 2 - second bit
+	setSwitchState(2, (cmd.value & 2)? "on" : "off")
+	//switch 3 - third bit
+	setSwitchState(3, (cmd.value & 4)? "on" : "off")
+	//switch 4 - fourth bit
+	setSwitchState(4, (cmd.value & 8)? "on" : "off")
+	//switch 5 - fifth bit
+	setSwitchState(5, (cmd.value & 16)? "on" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelStartLevelChange cmd) {
+	//Not supported
+	//We have no way to set and/or retrieve multilevel state of each button
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelStopLevelChange cmd) {
+	//Not supported
+	//We have no way to set and/or retrieve multilevel state of each switch
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationBusy cmd) {
+	//we have no way of knowing which command was ignored
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.scenecontrollerconfv1.SceneControllerConfReport cmd) {
+	if (cmd.groupId != cmd.sceneId) {
+		//Scene not set up properly for this association group. Try setting it two more times.
+		def cmds = []
+		cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfSet(dimmingDuration: 0, groupId: cmd.groupId, sceneId: cmd.groupId).format())
+		cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfSet(dimmingDuration: 0, groupId: cmd.groupId, sceneId: cmd.groupId).format())
+		sendHubCommand(cmds, 1500)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Unexpected zwave command $cmd"
+}
+
+//method created to make child class reusable
+void childOn(deviceNetworkId) {
+	def switchId = deviceNetworkId?.split("/")[1]
+	//this may override other local switch states if cloud is out of sync
+	updateLocalSwitchState()
+}
+
+//method created to make child class reusable
+void childOff(deviceNetworkId) {
+	def switchId = deviceNetworkId?.split("/")[1]
+	//this may override other local switch states if cloud is out of sync
+	updateLocalSwitchState()
+}
+
+private setSwitchState(switchId, state) {
+	String childDni = "${device.deviceNetworkId}/${switchId}"
+	def child = childDevices.find{it.deviceNetworkId == childDni}
+	if (!child) {
+		log.error "Child device $childDni not found"
+	}
+	//send event only if state changed
+	if(child?.device.currentState("switch")?.value != state) {
+		child?.sendEvent(name: "switch", value: "$state", descriptionText: "$child.displayName was switched $state")
+	}
+}
+
+private updateLocalSwitchState() {
+	def binarySwitchState = 0;
+	def multiplier = 1;
+	for (int i = 1; i <= 5; ++i) {
+		String childDni = "${device.deviceNetworkId}/${i}"
+		def child = childDevices.find{it.deviceNetworkId == childDni}
+		if (child?.device.currentState("switch")?.value == "on") {
+			binarySwitchState += multiplier
```

Comment:
Switch 1 should be bit 0, Switch 2 should be bit 1, etc

I think this will be reversed from how you'd want it to be, no?

Reply:
I am not quite certain what you mean. Could you please elaborate?

If you mean binarySwitchState value, that is how it works. For button 1 we get multiplier == 1, so will bit 0 (least significant bit, which in IndicatorReport I called first bit, which may be confusing :/ ) will be 1 if switch is "on" and 0 if it is "off" (and it is not quite our design, that is how device handles value for IndicationSet command).

If you mean ${i} index in childDni, we kept approach from existing Aeon Minimote, where switches are numbered from 1 to 4 in UI, and keep the same values in childDni (probably to avoid confusion when debugging). Of course we can change that if you want.

Reply:
Oh nope, you're right.

I read it wrong.